modules/unsupported/wfs-ng/src/main/java/org/geotools/data/wfs/internal/v2_0/StrictWFS_2_0_Strategy.java

```diff
@@ -588,55 +512,230 @@ protected String getOperationURI(WFSOperationType operation, HttpMethod method)
+
+        Set<String> serverSupportedFormats;
+        serverSupportedFormats = findParameters(operationMetadata, parameterName);
+        return serverSupportedFormats;
     }
 
     @Override
     public Set<String> getServerSupportedOutputFormats(QName typeName, WFSOperationType operation) {
-        // TODO Auto-generated method stub
-        return null;
+        Set<String> ftypeFormats = new HashSet<String>();
+
+        final Set<String> serviceOutputFormats = getServerSupportedOutputFormats(operation);
+        ftypeFormats.addAll(serviceOutputFormats);
+
+        if (GET_FEATURE.equals(operation)) {
+            final FeatureTypeInfo typeInfo = getFeatureTypeInfo(typeName);
+
+            final Set<String> typeAdvertisedFormats = typeInfo.getOutputFormats();
+
+            ftypeFormats.addAll(typeAdvertisedFormats);
+        }
+        return ftypeFormats;
     }
 
     @Override
     public List<String> getClientSupportedOutputFormats(WFSOperationType operation) {
-        // TODO Auto-generated method stub
-        return null;
+        List<WFSResponseFactory> operationResponseFactories;
+        operationResponseFactories = WFSExtensions.findResponseFactories(operation);
+
+        List<String> outputFormats = new LinkedList<String>();
+        for (WFSResponseFactory factory : operationResponseFactories) {
+            List<String> factoryFormats = factory.getSupportedOutputFormats();
+            outputFormats.addAll(factoryFormats);
+        }
+
+        if (GET_FEATURE.equals(operation)) {
+            for (String preferred : PREFERRED_FORMATS) {
+                boolean hasFormat = outputFormats.remove(preferred);
+                if (hasFormat) {
+                    outputFormats.add(0, preferred);
+                    break;
+                }
+            }
+        }
+
+        return outputFormats;
     }
 
     @Override
     public boolean supportsTransaction(QName typeName) {
-        // TODO Auto-generated method stub
-        return false;
+        try {
+            getFeatureTypeInfo(typeName);
+        } catch (IllegalArgumentException e) {
+            throw e;
+        }
+        if (!supportsOperation(TRANSACTION, POST)) {
+            return false;
+        }
+
+        /*OperationsType operations = this.capabilities.getFeatureTypeList().getOperations();
+        @SuppressWarnings("unchecked")
+        List<net.opengis.wfs.OperationType> operation = operations.getOperation();
+        for (net.opengis.wfs.OperationType required : Arrays.asList(
+                net.opengis.wfs.OperationType.INSERT_LITERAL,
+                net.opengis.wfs.OperationType.UPDATE_LITERAL,
+                net.opengis.wfs.OperationType.DELETE_LITERAL)) {
+
+            if (!operation.contains(required)) {
+                info("Transactions not supported since WFS didn't declare support for "
+                        + required.getName());
+                return false;
+            }
+        }*/
```

Comment:
The commented out code here makes a bit more sense to me, expect I am missing the point.